src/iuse.cpp

```diff
@@ -9009,25 +9173,42 @@ int iuse::washclothes( player *p, item *, bool, const tripoint & )
             return string_format( "%3d", val );
         };
         using stats = inventory_selector::stats;
-        return stats{{
+        return stats{ {
                 display_stat( _( "Water" ), required.water, available_water, to_string ),
                 display_stat( _( "Cleanser" ), required.cleanser, available_cleanser, to_string )
-            }};
+            } };
     };
-    // TODO: this should also search surrounding area, not just player inventory.
-    inventory_iuse_selector inv_s( *p, _( "ITEMS TO CLEAN" ), preset, make_raw_stats );
-    inv_s.add_character_items( *p );
-    inv_s.set_title( _( "Multiclean" ) );
-    inv_s.set_hint( _( "To clean x items, type a number before selecting." ) );
-    if( inv_s.empty() ) {
-        popup( std::string( _( "You have nothing to clean." ) ), PF_GET_KEY );
-        return 0;
-    }
-    std::list<std::pair<int, int>> to_clean = inv_s.execute();
-    if( to_clean.empty() ) {
-        return 0;
-    }
 
+    std::list<std::pair<int, int>> to_clean;
+    if( cbm ) {
+        // TODO: this should also search surrounding area, not just player inventory.
+        inventory_iuse_selector inv_cbm( *p, _( "ITEMS TO CLEAN" ), cbm_preset, make_raw_stats );
+        inv_cbm.add_character_items( *p );
+        inv_cbm.set_title( _( "Multiclean" ) );
+        inv_cbm.set_hint( _( "To clean x items, type a number before selecting." ) );
+        if( inv_cbm.empty() ) {
+            popup( std::string( _( "You have nothing to clean." ) ), PF_GET_KEY );
+            return 0;
+        }
+        to_clean = inv_cbm.execute();
+        if( to_clean.empty() ) {
+            return 0;
+        }
+    } else {
+        // TODO: this should also search surrounding area, not just player inventory.
+        inventory_iuse_selector inv_s( *p, _( "ITEMS TO CLEAN" ), preset, make_raw_stats );
+        inv_s.add_character_items( *p );
+        inv_s.set_title( _( "Multiclean" ) );
+        inv_s.set_hint( _( "To clean x items, type a number before selecting." ) );
+        if( inv_s.empty() ) {
+            popup( std::string( _( "You have nothing to clean." ) ), PF_GET_KEY );
+            return 0;
+        }
+        to_clean = inv_s.execute();
+        if( to_clean.empty() ) {
+            return 0;
+        }
+    }
```

Comment:
These almost identical code snippets ain't good. We need to somehow extract them to one generic function with arguments and apply it to both conditions.

Reply:
I tried that but the best I managed is to make wash_item() and have whashclothes() and washcbms() use it.